homeassistant/components/alarm_control_panel/group.py

```diff
@@ -0,0 +1,241 @@
+"""
+Group platform for alarm control panel component.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/alarm_control_panel.group/
+"""
+import asyncio
+from collections import Counter
+from copy import deepcopy
+import logging
+import voluptuous as vol
+
+from homeassistant.core import callback
+import homeassistant.components.alarm_control_panel as alarm
+from homeassistant.components.alarm_control_panel import (
+    DOMAIN, SERVICE_ALARM_DISARM, SERVICE_ALARM_ARM_HOME,
+    SERVICE_ALARM_ARM_AWAY, SERVICE_ALARM_ARM_NIGHT, SERVICE_ALARM_TRIGGER,
+    SERVICE_ALARM_ARM_CUSTOM_BYPASS)
+from homeassistant.const import (
+    ATTR_ENTITY_ID, ATTR_PRE_PENDING_STATE, ATTR_POST_PENDING_STATE,
+    STATE_ALARM_DISARMED, STATE_ALARM_ARMED_CUSTOM_BYPASS, STATE_ALARM_PENDING,
+    STATE_ALARM_TRIGGERED, CONF_PLATFORM, CONF_NAME, EVENT_HOMEASSISTANT_START)
+import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.event import async_track_state_change
+
+ATTR_TRIGGERED_PANELS = 'triggered_panels'
+
+CONF_CODE_FORMAT = 'code_format'
+CONF_PANELS = 'panels'
+CONF_PANEL = 'panel'
+
+DEFAULT_ALARM_NAME = 'Group Alarm'
+
+PLATFORM_SCHEMA = vol.Schema(vol.All({
+    vol.Required(CONF_PLATFORM): 'group',
+    vol.Optional(CONF_NAME, default=DEFAULT_ALARM_NAME): cv.string,
+    vol.Optional(CONF_CODE_FORMAT, default=''): cv.string,
+    vol.Required(CONF_PANELS): vol.All(cv.ensure_list, [{
+        vol.Required(CONF_PANEL): cv.entity_id,
+    }])
+}))
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the manual alarm platform."""
+    add_devices([GroupAlarm(
+        hass,
+        config[CONF_NAME],
+        config[CONF_CODE_FORMAT],
+        config.get(CONF_PANELS)
+        )])
+
+
+class GroupAlarm(alarm.AlarmControlPanel):
+    """Implement the notification service for the group notify platform."""
+
+    def __init__(self, hass, name, code_format, entities):
+        """Initialize the service."""
+        self._hass = hass
+        self._name = name
+        self._code_format = code_format if code_format else None
+        self._entity_ids = [entity.get(CONF_PANEL) for entity in entities]
+        self._triggered_panels = []
+        self._state = STATE_ALARM_DISARMED
```

Comment:
Why init to `STATE_ALARM_DISARMED`? I suggest init to `None`. See comment below about unknown state.